packages/backend/src/core/collectors/ValidiumStateTransitionCollector.test.ts

```diff
@@ -0,0 +1,216 @@
+import { EthereumAddress, Hash256 } from '@explorer/types'
+import { expect, mockFn } from 'earljs'
+
+import { BlockRange } from '../../model'
+import { StateTransitionRepository } from '../../peripherals/database/StateTransitionRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { HackFilter } from '../../peripherals/ethereum/HackJsonRpcProvider'
+import { mock } from '../../test/mock'
+import { LogRootUpdate, LogStateTransitionFact, LogUpdateState } from './events'
+import { ValidiumStateTransitionCollector } from './ValidiumStateTransitionCollector'
+
+describe(ValidiumStateTransitionCollector.name, () => {
+  describe(ValidiumStateTransitionCollector.prototype.discardAfter.name, () => {
+    it('discards all records from pageMappingRepository after given block', async () => {
+      const stateTransitionRepository = mock<StateTransitionRepository>({
+        deleteAfter: async () => 0,
+      })
+
+      const collector = new ValidiumStateTransitionCollector(
+        mock<EthereumClient>(),
+        stateTransitionRepository,
+        EthereumAddress.fake(),
+        LogRootUpdate
+      )
+
+      await collector.discardAfter(123)
+
+      expect(stateTransitionRepository.deleteAfter).toHaveBeenCalledWith([123])
+    })
```

Comment:
I don't know if the same test with `LogUpdateState` instead of `LogRootUpdate` would be useful

Reply:
I think it's fine to have a single test